ui/app/src/components/PreviewSearchPanel/PreviewSearchPanel.tsx

```diff
@@ -124,13 +130,17 @@ export function PreviewSearchPanel() {
   });
   const dispatch = useDispatch();
   const editMode = useSelection((state) => state.preview.editMode);
+  const allowedTypesData = useSelection((state) => state.preview.guest?.allowedContentTypes);
+  const awaitingGuestCheckIn = nou(allowedTypesData);
   const contentTypes = useContentTypeList(
     (contentType) => contentType.id !== '/component/level-descriptor' && contentType.type === 'component'
   );
   const contentTypesLookup = useMemo(
     () => (contentTypes ? createLookupTable(contentTypes, 'id') : null),
     [contentTypes]
   );
+  const onPageChangedSubscription = useRef<Subscription>();
+  const onRowsPerPageChangeSubscription = useRef<Subscription>();
```

Comment:
Why 2 subscriptions? Don't see why there would ever be two active. Also, the unsubscribe needs to be called before recreating the new subscription.

**Check if calling unsubscribe on completed subscriptions would throw**. If so, you need to clear the ref after unsubscribing.

You may even use the same ref for the useEffect.

Reply:
After cheking, calling unsubscribe on completed subscriptions doesn't throw an error), it doesn't do anything.